src/Integration/WooCommercePreOrders.php

```diff
@@ -0,0 +1,71 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Integration;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\WCProductAdapter;
+use WC_Pre_Orders_Product;
+use WC_Product;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WooCommercePreOrders
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Integration
+ *
+ * @link https://woocommerce.com/products/woocommerce-pre-orders/
+ *
+ * @since x.x.x
+ */
+class WooCommercePreOrders implements IntegrationInterface {
+
+	/**
+	 * Returns whether the integration is active or not.
+	 *
+	 * @return bool
+	 */
+	public function is_active(): bool {
+		return defined( 'WC_PRE_ORDERS_VERSION' );
+	}
+
+	/**
+	 * Initializes the integration (e.g. by registering the required hooks, filters, etc.).
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		add_filter(
+			'woocommerce_gla_product_attribute_values',
+			function ( array $attributes, WC_Product $product ) {
+				return $this->maybe_set_preorder_availability( $attributes, $product );
+			},
+			2,
+			10
+		);
+	}
+
+	/**
+	 * Sets the product's availability to "preorder" if it's in-stock and can be pre-ordered.
+	 *
+	 * @param array      $attributes
+	 * @param WC_Product $product
+	 *
+	 * @return array
+	 */
+	protected function maybe_set_preorder_availability( array $attributes, WC_Product $product ): array {
+		if ( $product->is_in_stock() && WC_Pre_Orders_Product::product_can_be_pre_ordered( $product ) ) {
+			$attributes['availability'] = WCProductAdapter::AVAILABILITY_PREORDER;
+
+			$timestamp = WC_Pre_Orders_Product::get_localized_availability_datetime_timestamp( $product );
+			if ( ! empty( $timestamp ) ) {
+				// Convert the timestamp back to UTC.
+				// The localized offset is already applied to timestamp by WC_Pre_Orders_Product.
+				$utc_timestamp                  = $timestamp - wc_timezone_offset();
+				$attributes['availabilityDate'] = gmdate( 'c', intval( $utc_timestamp ) );
```

Comment:
Good point. This was added mainly to counter act a similar method used in Pre-orders extension to return the localize time but I see that these two methods are slightly different.

I initially hesistated to use the meta key because didn't want the integration to be tightly coupled with the extension implementation but I guess even if we use `wc_timezone_string` to convert the timezone, it would still mean we're relying on `get_localized_availability_datetime_timestamp`'s implementation.

I changed the code to use the `_wc_pre_orders_availability_datetime` meta value for getting the availability datetime in https://github.com/woocommerce/google-listings-and-ads/pull/959/commits/e6e576aa126bc1353cb193ff2b692e13bb396ed3